app/src/main/java/org/p2p/wallet/common/ui/widget/EarnWidget.kt

```diff
@@ -0,0 +1,183 @@
+package org.p2p.wallet.common.ui.widget
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.widget.FrameLayout
+import androidx.annotation.ColorRes
+import androidx.annotation.StringRes
+import androidx.core.view.isVisible
+import com.robinhood.ticker.TickerUtils
+import com.robinhood.ticker.TickerView
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.inject
+import org.p2p.uikit.glide.GlideManager
+import org.p2p.uikit.utils.getColor
+import org.p2p.uikit.utils.toPx
+import org.p2p.wallet.R
+import org.p2p.wallet.databinding.ItemDepositTokenBinding
+import org.p2p.wallet.databinding.WidgetEarnViewBinding
+import org.p2p.wallet.utils.DecimalFormatter
+import org.p2p.wallet.utils.isZero
+import org.p2p.wallet.utils.viewbinding.inflateViewBinding
+import java.math.BigDecimal
+
+private const val DEFAULT_BUTTON_TEXT_PADDING_DP = 20
+private const val SMALL_BUTTON_TEXT_PADDING_DP = 16
+
+private const val TOKEN_MARGIN_END_DP = 8
+
+private const val TICKER_DECIMALS = 12
+
+class EarnWidget @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0
+) : FrameLayout(context, attrs, defStyleAttr), KoinComponent {
+
+    private val binding: WidgetEarnViewBinding = inflateViewBinding()
+
+    private val glideManager: GlideManager by inject()
+
+    var currentState: EarnWidgetState? = null
```

Comment:
I think widget can't have NULL state even at init time. Maybe it will be more convinient to have DEFAULT state or smth like that? 
null is evil >:(

Reply:
`EarnWidgetState.IDLE`

Reply:
and what to show in IDLE state ? just background with button show deposits ?